include/service/entities/plugins/plugin_initialization_service_v2.js

```diff
@@ -0,0 +1,150 @@
+const Promise = require('bluebird');
+const fs = require('fs');
+const path = require('path');
+
+Promise.promisifyAll(fs);
+
+module.exports = (pb) => {
+    const DETAILS_FILE_NAME = 'details.json';
+    const PLUGINS_DIR = path.join(pb.config.plugins.directory);
+    const PUBLIC_DIR_NAME = 'public';
+    const util = pb.util;
+
+    class PluginInitializationService {
+        constructor (pluginuid) {
+            this.pluginuid = pluginuid;
+        }
+
+        static handleInitializationError (pluginuid, err) {
+            //log it all
+            var hasValidationErrs = !!err.validationErrors;
+            pb.log.error('PluginInitializationService:[%s] failed to initialize: %s', pluginuid, hasValidationErrs ? err.message : err.stack);
+            if (hasValidationErrs) {
+                err.validationErrors.forEach(function(validationError) {
+                    pb.log.error('PluginInitializationService:[%s] details.json validation error FIELD=%s MSG=%s', pluginuid, validationError.field, validationError.message);
+                });
+            }
+        }
+
+        initialize() {
+            return this._getDetails()
+                .then(details => this._validate(details))
+                .then(details => this._initTasks(details));
+        }
+
+        _initTasks(details) {
+            const mainModule = this._loadMainModule(details);
+            let pluginSpec = this._mainModuleHandler(mainModule, details);
+            pluginSpec.uid = this.pluginuid;
+
+            return this._onStartup(pluginSpec.main_module)
+                .then(_ => this._loadServices())
+                .then(services => pluginSpec.services = services)
+                .then(_ => this._loadControllers())
+                .then(controllers => {
+                    pluginSpec.controllers = controllers
+                    this._loadLocalization();
+                    return pluginSpec;
+                });
+        }
+
+        _loadMainModule(details) {
+            try {
+                return require(path.join(PLUGINS_DIR, this.pluginuid, details.main_module.path))(pb);
+            }
+            catch (e) {
+                return null;
+            }
+        }
+
+
+        // Platform Wide Tasks (Only need to run once)
+        _getDetails() {
+            const filePath = path.join(PLUGINS_DIR, this.pluginuid, DETAILS_FILE_NAME);;
+            return fs.readFileAsync(filePath).then((data) => {
+                try {
```

Comment:
now that we're in a promise handler we can just replace this with `fs.readFileAsync(filePath).then(JSON.parse)` and the error will be handled normally.